### ISSUE TITLE
Fingerprint.com

### ISSUE THREAD

**Issue by Stryder24** [kkapsner/CanvasBlocker#643]
The Fingerprint Pro Demo (https://fingerprint.com/demo/), is able to determine a unique identifier for my browser.  I can change my IP Address by connecting to different locations and that does not change the identifier.   The browser identity remained the same if I opened incognito windows.

I was wondering if there was a way to configure Canvas Blocker to successfully avoid being fingerprinted.    They claim that they can Identify anonymous site visitors with 99.5% accuracy to "prevent online fraud"

Windows 10
Firefox 64bit 110.0
Canvas Blocker 1.8
UBlock Origin
privacy.resistFingerprinting enabled
the arkenfox user.js
Default Canvas Blocker configuration

Thank you

**Comment by kkapsner** [comment]
I briefly  checked and it seems they store some cookie/local storage/similar in your browser.

When I open two Firefox instances with different (fresh) profiles the identifier changes:
![grafik](https://user-images.githubusercontent.com/1105157/226901814-c1137b6a-c8ea-4ea6-a860-37a90a30ec00.png)

So I would say CanvasBlocker is working there. To deal with the persistent "cookie" (I have not looked into the details of what they really store) you have to look into your cookie settings or other extensions that deal with that.

**Comment by Thorin-Oakenpants** [comment]
FYI: read this thread and listen to what Peter Snyder (head of Brave privacy etc) says - https://github.com/brave/brave-browser/issues/14031. This has also been discussed at `https://github.com/arkenfox/user.js/issues/1647` and without going into detail (and it's been discussed at tor project in the past), it's all misleading (intentionally or not) to those who don't understand how FPing works

tl;dr: the best you can do without a crowd, e.g. FF or Brave, is randomize and fool naive scripts. This still doesn't cover sanitizing state tracking, eliminating navigational tracking, and last but not least, your IP (as a loose fuzzy data point, or a precise one). The science behind linking FPs behind the scenes is another whole area.

I wouldn't call this script advanced - it doesn't seem to take into account randomizing, and I can easily fool it by just changing some measurements - metrics can be collected, but they should be stable in linkfying profiles. They are also storing state and using IP - which to be fair, is fair game - but this is not entirely fingerprinting on it's own

As shown in numerous anecdotes: RFP (with random canvas), and sanitizing is enough to fool it in the web demo - behind the scenes may be a different story (but I doubt it). They only care about low hanging fruit and no false positives - so a combo of useragent and IP is almost enough to nab 99% of people. Don't sweet it, it's not that advanced

**Comment by Stryder24** [comment]
Ok, thank you for letting me know.   Within the limits of my knowledge and experience, this seemed like something new and more effective than previous methods.  The information above is very helpful.  Have a great day.
